sdk/rust-examples/huffman-encoding/src/main.rs

```diff
@@ -0,0 +1,161 @@
+//! Huffman Encoding Example
+//!
+//! ## Context
+//!
+//! Reads user input, and then encodes the file to binary format to compress it and save space.
+//! And then, it prints the encoded file.
+//! After that, the encoded file gets converted back to the original text, then that is outputted.
+//! WARNING: Program is only desinged for ASCII characters, not UTF-8 encoding !!
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Copyright
+//!
+//! See the file `LICENSING.markdown` in the Veracruz root directory for licensing and
+//! copyright information.
+
+use anyhow::{self, Ok};
+use std::{collections::HashMap, fs};
+
+const INPUT_FILENAME: &'static str = "/input/huffman_example_input.txt";
+const OUTPUT_FILENAME: &'static str = "/output/encoded_output.txt";
+
+type Link = Option<Box<Node>>;
+
+#[derive(Debug)]
+
+/// A Binary Tree Node is represented here
+struct Node {
+    /// Each Node, must represent a character, and its frequency in the input string
+    freq: i32,
```

Comment:
Can a frequency ever be negative?  If the answer is "no", maybe it makes sense to represent frequencies using an unsigned type, like `u32`?